lib/presentation/util/extensions/upload_request_status_extension.dart

```diff
@@ -0,0 +1,68 @@
+// LinShare is an open source filesharing software, part of the LinPKI software
+// suite, developed by Linagora.
+//
+// Copyright (C) 2020 LINAGORA
+//
+// This program is free software: you can redistribute it and/or modify it under the
+// terms of the GNU Affero General Public License as published by the Free Software
+// Foundation, either version 3 of the License, or (at your option) any later version,
+// provided you comply with the Additional Terms applicable for LinShare software by
+// Linagora pursuant to Section 7 of the GNU Affero General Public License,
+// subsections (b), (c), and (e), pursuant to which you must notably (i) retain the
+// display in the interface of the “LinShare™” trademark/logo, the "Libre & Free" mention,
+// the words “You are using the Free and Open Source version of LinShare™, powered by
+// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
+// offer!”. You must also retain the latter notice in all asynchronous messages such as
+// e-mails sent with the Program, (ii) retain all hypertext links between LinShare and
+// http://www.linshare.org, between linagora.com and Linagora, and (iii) refrain from
+// infringing Linagora intellectual property rights over its trademarks and commercial
+// brands. Other Additional Terms apply, see
+// <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf>
+// for more details.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+
+
+import 'package:domain/domain.dart';
+import 'package:flutter/widgets.dart';
+import 'package:linshare_flutter_app/presentation/localizations/app_localizations.dart';
+import 'package:linshare_flutter_app/presentation/util/extensions/color_extension.dart';
+
+extension UploadRequestStatusDisplayExtension on UploadRequestStatus {
+  String displayValue(BuildContext context) {
+    switch(this) {
+      case UploadRequestStatus.ENABLED:
+        return AppLocalizations.of(context).active;
+      case UploadRequestStatus.ARCHIVED:
+        return AppLocalizations.of(context).archived;
+      case UploadRequestStatus.CLOSED:
+        return AppLocalizations.of(context).expired_closed;
+      case UploadRequestStatus.CREATED:
+        return AppLocalizations.of(context).pending;
+      default:
+        return '';
+    }
+  }
+
+  Color get displayColor {
```

Comment:
display color for what?

Reply:
this is extension which get `display color` for `Upload Request Status`

Reply:
yes, please define that for the method name, it should be make sense for re-use by other developer. With good name, we don't need comment on code.

Reply:
LOL, you may not the point. The `display` here is a noun I mean, not is a verb. And I wrote this function in `upload_request_status_extension`, so others can understand

Reply:
LOL, you may not get the point. The `display` here is a noun I mean, not is a verb. And I wrote this function in `upload_request_status_extension`, so others can understand

Reply:
hmm, srr, I miss the context: it is an extension. OK with it